core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```
@@ -97,14 +99,16 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
 
     loadCache()
 
-    aclChangeListener = new ZkNodeChangeNotificationListener(zkClient, AclChangeNotificationZNode.path, AclChangeNotificationSequenceZNode.SequenceNumberPrefix, AclChangedNotificationHandler)
+    aclChangeListener = new ZkNodeChangeNotificationListener(zkClient, AclStore.literalAclStore.aclChangesZNode.path, AclStore.literalAclStore.aclChangeNotificationSequenceZNode.SequenceNumberPrefix, AclChangedNotificationHandler)
     aclChangeListener.init()
+    wildcardSuffixedAclChangeListener = new ZkNodeChangeNotificationListener(zkClient, AclStore.wildcardSuffixedAclStore.aclChangesZNode.path, AclStore.wildcardSuffixedAclStore.aclChangeNotificationSequenceZNode.SequenceNumberPrefix, AclChangedNotificationHandler)
+    wildcardSuffixedAclChangeListener.init()
   }
 
   override def authorize(session: Session, operation: Operation, resource: Resource): Boolean = {
     val principal = session.principal
     val host = session.clientAddress.getHostAddress
-    val acls = getAcls(resource) ++ getAcls(new Resource(resource.resourceType, Resource.WildCardResource))
+    val acls = getMatchingAcls(resource)
```

should throw an error if `resource` is not a `Literal`, i.e. concrete, resource. (Probably best to put the change in `getMatchingAcls` if this is going to be public.